### PR TITLE
added organisations requests page with filtrable list of all appliations

### PR DESCRIPTION
**Pull request by blueQcloud**

### CODE REVIEW

File: app/containers/Admin/OrganisationRequestsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      <Table
+        striped
+        hover
+        style={{textAlign: 'center', border: '1px solid #f5f5f5'}}
+      >
+        <thead style={{backgroundColor: '#036', color: 'white'}}>
+          <tr>
+            <th>#</th>
+            <th>Name</th>
+            <th>Lastname</th>
```
Comment (matthieu-foucault) [inline app/containers/Admin/OrganisationRequestsTable.tsx]: We should have the user email in this table. Name and last name are not enough to identify a user

File: app/containers/Admin/OrganisationRequestsTableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import {graphql, createFragmentContainer} from 'react-relay';
+import {OrganisationRequestsTableRow_userOrganisation} from '__generated__/OrganisationRequestsTableRow_userOrganisation.graphql';
+interface Props {
+  userOrganisation: OrganisationRequestsTableRow_userOrganisation;
+  key: string;
+}
+export const OrganisationRequestsTableRowComponent: React.FunctionComponent<
+  Props
+> = props => {
+  console.log(props);
```
Comment (matthieu-foucault) [inline app/containers/Admin/OrganisationRequestsTableRow.tsx]:
```suggestion
```

File: app/containers/Organisations/UserOrganisation.tsx [hunk 3]
```diff
@@ -11,10 +11,10 @@ export const UserOrganisationComponent = props => {
...
-    attention: 'warning',
+    rejected: 'warning',
     pending: 'info',
-    expired: 'danger',
-    active: 'success'
+    // Expired: 'danger',
+    approved: 'success'
   };
 
   /** TODO: INTENDED FOR DEMO ONLY. To be removed */
```
Comment (matthieu-foucault) [inline app/containers/Organisations/UserOrganisation.tsx]: To properly the new approval interface, while keeping this behavior for development, I suggest that we create two buttons: one that sets directly sets the status as active, only visible in development (`process.env.NODE_ENV !== 'production'`) and one that creates a pending application.

File: app/server/schema.graphql [hunk 4]
```diff
@@ -6918,9 +6947,7 @@ input UserOrganisationCondition {
 input UserOrganisationInput {
   """foreign key that references the organisation table"""
   organisationId: Int!
-
-  """the status of user access to the organisation"""
-  status: String!
+  status: Statuses
```
Comment (matthieu-foucault) [inline app/server/schema.graphql]: :heart_eyes:

File: schema/deploy/type_statuses.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+-- Deploy ggircs-portal:type_statuses to pg
+-- requires: schema_ggircs_portal
+
+begin;
+
+create type ggircs_portal.statuses as enum ('pending', 'approved', 'rejected');
```
Comment (matthieu-foucault) [inline schema/deploy/type_statuses.sql]: The name of this type should be more precise. `user_organisation_status` is better, as it tells us in which context the type is used

File: app/containers/Organisations/UserOrganisation.tsx (hunk 3 above)
Comment (dleard) [reply app/containers/Organisations/UserOrganisation.tsx]: good idea

File: schema/deploy/type_statuses.sql (hunk 5 above)
Comment (dleard) [reply schema/deploy/type_statuses.sql]: :+1:

**Pull request closed without merging**